profile_queries.sh

```diff
@@ -183,11 +183,65 @@ has_profile() {
         echo "The query contains the word 'PROFILE' for transaction ID '$TXN_ID' and statement ID '$STMT_ID':"
     else
         echo "ERROR: The query does not contain the word 'PROFILE' for transaction ID '$TXN_ID' and statement ID '$STMT_ID'"
-        echo "All the queries should contain the word 'PROFILE' in it."
+        echo "All the queries should contain the word 'PROFILE' in them."
         exit 1
     fi
 }
 
+# Checking retention issues if transactions are provided
+if [ -n "$transactions" ]; then
+    condition=""
+    for (( i=0; i<${#TXN_IDS[@]}; i++ )); do
+        if [[ $i -ne 0 ]]; then
+            condition+=" OR "
+        fi
+        condition+="(transaction_id = ${TXN_IDS[i]} AND statement_id = ${STMT_IDS[i]})"
+    done
+    # Get the minimum start timestamp based on the condition
+    query_min_time=$($VSQL_ADMIN_COMMAND -Atc "SELECT MIN(start_timestamp) FROM query_requests WHERE $condition;")
+    echo "Query minimum start time: $query_min_time"
```

Comment:
Suggest a more informative/truthful message not only dealing with plurals (minimum) but making clear that this is the minimum of ALL queries provided:
if [ $i -eq 0 ]; then
  echo "Query start time: $query_min_time"
else
  echo "Minimum of all queries start time: $query_min_time"
fi

Reply:
Good idea!